ethcore/src/executive.rs

```diff
@@ -977,11 +977,18 @@ impl<'a, B: 'a + StateBackend> Executive<'a, B> {
 		if stack_depth != depth_threshold {
 			self.call_with_stack_depth(params, substate, stack_depth, tracer, vm_tracer)
 		} else {
-			crossbeam::scope(|scope| {
-				scope.builder().stack_size(::std::cmp::max(self.schedule.max_depth.saturating_sub(depth_threshold) * STACK_SIZE_PER_DEPTH, local_stack_size)).spawn(move || {
-					self.call_with_stack_depth(params, substate, stack_depth, tracer, vm_tracer)
-				}).expect("Sub-thread creation cannot fail; the host might run out of resources; qed")
-			}).join().expect("Sub-thread never panics; qed")
+			thread::scope(|scope| {
+				let stack_size = cmp::max(self.schedule.max_depth.saturating_sub(depth_threshold) * STACK_SIZE_PER_DEPTH, local_stack_size);
+				scope.builder()
+					.stack_size(stack_size)
+					.spawn(|_| {
+						self.call_with_stack_depth(params, substate, stack_depth, tracer, vm_tracer)
+					})
+					.expect("Sub-thread creation cannot fail; the host might run out of resources; qed")
+					.join()
+			})
+			.expect("Sub-thread never panics; qed")
+			.expect("Sub-thread never panics; qed")
```

Comment:
crossbeamapi has changed a bit here,
- We have to call `join` inside `fn scope` callback, cause the `join` handle lifetime is now the same as scope lifetime :(
- `fn scope` returns error, so that's why we have double `expect` here

Reply:
> fn scope returns error, 

What are the conditions under which it errors?

Reply:
> If the child thread panics, an error is returned.

from https://docs.rs/crossbeam/0.7.1/crossbeam/thread/struct.ScopedJoinHandle.html

and

> If any of the joined threads has panicked, an Err is returned containing errors from panicked threads.

from https://docs.rs/crossbeam/0.7.1/crossbeam/fn.scope.html